src/tracking.c

```diff
@@ -99,6 +99,43 @@ void disableTracking(client *c) {
     }
 }
 
+static int stringCheckPrefix(unsigned char *s1, size_t s1_len, unsigned char *s2, size_t s2_len) {
+    size_t min_length = s1_len < s2_len ? s1_len : s2_len;
+    return memcmp(s1,s2,min_length) == 0;   
+}
+
+/* Check if any of the provided prefixes collide with one another or
+ * with an existing prefix. A collision is defined as two prefixes
+ * that will emit a invalidation for overlapping subsets of keys. If
+ * a prefix is found, the index of the conflicting prefix is returned,
+ * otherwise -1 is returned. */
+int findPrefixCollisions(client *c, robj **prefixes, size_t numprefix) {
+    for (size_t i = 0; i < numprefix; i++) {
+        /* Check input list has no overlap with existing prefixes. */
+        if (c->client_tracking_prefixes) {
+            raxIterator ri;
+            raxStart(&ri,c->client_tracking_prefixes);
+            raxSeek(&ri,"^",NULL,0);
+            while(raxNext(&ri)) {
+                if(stringCheckPrefix(ri.key,ri.key_len,
+                        prefixes[i]->ptr,sdslen(prefixes[i]->ptr))) {
+                    raxStop(&ri);
+                    return i;
+                }
+            }
+            raxStop(&ri);
+        }
+        /* Check input has no overlap with itself. */
+        for (size_t j = i + 1; j < numprefix; j++) {
```

Comment:
@madolson I felt like putting this check separately before the existing prefixes check would save some performance if user provided duplicate prefix in the same call, since I assuming the number of prefixes provided for single call should be much smaller than the overall, and this will make the check fail early.  maybe code looks like little bit redundant, how do you think?

Reply:
the chance of a user providing duplicate prefixes in the same call is smaller IMHO.

Reply:
I agree with Oran, I thought there was a pretty small chance of providing duplicates within the existing command.